### PR TITLE
Fixes in computing function endpoints for replication cases

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

https://jira.autodesk.com/browse/DYN-1759

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@reddyashish 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/Engine/ProtoTest/TD/Associative/Replication.cs [hunk 1]
```diff
@@ -4990,6 +4990,63 @@ public void TestReplicationWithArraysOfDifferentRanks()
...
+        {
+            string code = @"
+def foo ( a : double[], b :double[] )
+{
+    return = Count(a) + Count(b);
+}
+a = [ 1, 2 ];
+b = [[3, 4], [5,9]];
+c = b[[true,0]][0..1]; // [[b[true][0], b[true, 1]], [b[0,0], b[0,1]]] => [null, [3, 4]]
+test = foo (c, a[0..1]);	";
```
Comment (reddyashish) [inline test/Engine/ProtoTest/TD/Associative/Replication.cs]: Minor change, extra spaces here at the end of the string.

**Comment by reddyashish** [comment]
@aparajit-pratap Looks good to me.

File: test/Engine/ProtoTest/TD/Associative/Replication.cs [hunk 2]
```diff
@@ -4990,6 +4990,63 @@ public void TestReplicationWithArraysOfDifferentRanks()
...
+        {
+            string code = @"
+def foo ( a : double[], b :double[] )
+{
+    return = Count(a) + Count(b);
+}
+a = [ 1, 2 ];
+b = [[3, 4], [5,9]];
+c = b[[0.1,1.1]][0..1];
+test = foo (c, a[0..1]);";
```
Comment (mjkkirschner) [inline test/Engine/ProtoTest/TD/Associative/Replication.cs]: isn't  `a[0..1]` just equal to `a`

Comment (reddyashish) [reply test/Engine/ProtoTest/TD/Associative/Replication.cs]: yes they are same.

**Comment by mjkkirschner** [comment]
LGTM

**Comment by QilongTang** [comment]
@aparajit-pratap @reddyashish @mjkkirschner Merged

**Pull request merged**